core/src/main/java/org/apache/iceberg/rest/ScanTaskIterable.java

```diff
@@ -143,10 +141,14 @@ public void run() {
         int remaining = activeWorkers.decrementAndGet();
 
         if (remaining == 0
-            && !planTasks.isEmpty()
+            && planTasks.isEmpty()
```

Comment:
actually...if there is an explicit shutdown, I think we'll want to also insert the poison pill as well.